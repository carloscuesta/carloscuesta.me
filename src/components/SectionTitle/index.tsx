import Link from 'next/link'

type Props = {
  subTitle?: string
  title: string
  viewAllLink?: string
}

const SectionTitle = (props: Props) => (
  <div className="my-6 flex items-end justify-between">
    <div className={props.subTitle ? '' : 'space-y-2'}>
      <h2 className="text-lg font-bold">{props.title}</h2>
      {props.subTitle && (
        <h3 className="text-md opacity-70">{props.subTitle}</h3>
      )}
    </div>

    {props.viewAllLink && (
      <Link
        className="text-sm opacity-70 transition-opacity hover:opacity-100"
        href={props.viewAllLink}
      >
        View all
      </Link>
    )}
  </div>
)

export default SectionTitle
