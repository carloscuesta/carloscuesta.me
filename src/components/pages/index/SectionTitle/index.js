// @flow
import React from 'react'

import styles from './styles.module.css'

type Props = {
  title: string,
  subTitle: string
}

const SectionTitle = (props: Props) => (
  <>
    <h3 className={styles.sectionTitle}>{props.title}</h3>
    <h5 className={styles.sectionSubtitle}>{props.subTitle}</h5>
  </>
)

export default SectionTitle
