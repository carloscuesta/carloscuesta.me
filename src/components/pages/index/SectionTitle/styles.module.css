.sectionTitle {
  font-size: 2em;
  text-align: center;
  margin: 0;
}

.sectionSubtitle {
  color: var(--textSecondary);
  font-weight: 300;
  text-align: center;
  margin: 0;
  margin-bottom: 1.5em;
}
