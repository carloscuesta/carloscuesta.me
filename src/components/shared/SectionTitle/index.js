// @flow
import React, { type Node } from 'react'

import styles from './styles.module.css'

type Props = {
  isCentered?: boolean,
  subTitle?: string,
  title: string
}

const SectionTitle = (props: Props): Node => (
  <>
    <h4
      className={[
        styles.sectionTitle,
        props.isCentered ? styles.isCentered : undefined,
        !props.subTitle ? styles.sectionTitleMargin : undefined
      ].join(' ')}
    >
      {props.title}
    </h4>
    {props.subTitle &&
      <h5 className={[styles.sectionSubtitle, props.isCentered && styles.isCentered].join(' ')}>
        {props.subTitle}
      </h5>}
  </>
)

export default SectionTitle
