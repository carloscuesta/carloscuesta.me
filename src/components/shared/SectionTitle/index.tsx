import Link from 'next/link'

import styles from './styles.module.css'

type Props = {
  subTitle?: string,
  title: string,
  viewAllLink?: string
}

const SectionTitle = (props: Props) => (
  <div
    className={[
      styles.container,
      props.viewAllLink ? styles.containerWithLink : undefined
    ].join(' ')}
  >
    <div>
      <h2
        className={[
          styles.sectionTitle,
          !props.subTitle ? styles.sectionTitleMargin : undefined
        ].join(' ')}
      >
        {props.title}
      </h2>
      {props.subTitle &&
        <h3 className={styles.sectionSubtitle}>
          {props.subTitle}
        </h3>}
    </div>

    {props.viewAllLink &&
      <Link href={props.viewAllLink}>
        View all
      </Link>}
  </div>
)

export default SectionTitle
