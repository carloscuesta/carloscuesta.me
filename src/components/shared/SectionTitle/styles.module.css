.container {
  margin-bottom: 1.5rem;
}

.containerWithLink {
  align-items: flex-end;
  display: flex;
  justify-content: space-between;
}

.sectionTitle {
  font-size: 2rem;
  margin: 0;
  font-weight: 700;
}

.sectionTitleMargin {
  margin: 1rem 0;
}

.sectionSubtitle {
  color: var(--textSecondary);
  font-weight: 400;
  margin: 0;
  font-size: 1.25rem;
}

@media (max-width: 768px) {
  .container {
    align-items: flex-start;
    flex-direction: column;
  }

  .sectionSubtitle {
    margin-bottom: .5rem;
  }
}
