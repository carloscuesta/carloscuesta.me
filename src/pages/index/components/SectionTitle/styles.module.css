.sectionTitle {
  font-size: 2em;
  text-align: center;
}

.sectionSubtitle {
  color: var(--textSecondary);
  font-weight: 300;
  text-align: center;
  margin-bottom: 1.5em;
}
