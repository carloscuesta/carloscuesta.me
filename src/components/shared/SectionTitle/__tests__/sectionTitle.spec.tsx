import renderer from 'react-test-renderer'

import SectionTitle from '../index'
import * as stubs from './stubs'

describe('SectionTitle', () => {
  describe('with title and subtitle', () => {
    it('should match SectionTitle component', () => {
      const wrapper = renderer.create(<SectionTitle {...stubs.props} />)

      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('with title, subtitle and viewAllLink', () => {
    it('should match SectionTitle component', () => {
      const wrapper = renderer.create(<SectionTitle {...stubs.propsWithLink} />)

      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('with title only', () => {
    it('should match SectionTitle component', () => {
      const wrapper = renderer.create(<SectionTitle title='Test' />)

      expect(wrapper).toMatchSnapshot()
    })
  })
})
