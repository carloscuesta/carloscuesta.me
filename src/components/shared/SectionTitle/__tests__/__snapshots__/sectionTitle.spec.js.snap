// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionTitle should match SectionTitle component 1`] = `
Array [
  <h4
    className="sectionTitle  "
  >
    Title
  </h4>,
  <h5
    className="sectionSubtitle "
  >
    Subtitle
  </h5>,
]
`;
